docs/testing.md

```diff
@@ -39,20 +39,54 @@ They generally follow TPCH but are not guaranteed to be the same.
 
 ## Unit tests
 
-Unit tests exist in the tests directory. This is unconventional and is done so we can run the tests
-on the final shaded version of the plugin. It also helps with how we collect code coverage.
+Unit tests exist in the [tests](../tests) directory. This is unconventional and is done so we can
+ run the tests on the final shaded version of the plugin. It also helps with how we collect code coverage.
+
+In order to run the unit-tests issue the maven command to run the tests i.e. `mvn test`. 
+
+To run individual tests append `-DwildcardSuites=<comma separated list of wildcard suite names to execute>` to the above command 
+
+For more information about using scalatest with maven please refer [here](https://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin)
+    
+#### Running unit-tests against specific apache spark versions. 
 You can run the unit tests against different versions of Spark using the different profiles. The
-default version runs again Spark 3.0.0, to run against other version use one of the following profiles:
-   - `-Pspark301tests` (Spark 3.0.1)
-   - `-Pspark302tests` (Spark 3.0.2)
-   - `-Pspark310tests` (Spark 3.1.0)
+default version runs against Spark 3.0.0, to run against other versions use one of the following
+ profiles:
+   - `-Pspark301tests` (spark 3.0.1)
+   - `-Pspark302tests` (spark 3.0.2)
+   - `-Pspark310tests` (spark 3.1.0)
+
+Please refer to the [pom.xml](tests/pom.xml) to see the list of profiles supported
+apache spark specific configurations can be passed in by setting environment-variable spark_conf 
+
+Examples: 
+-To run tests against Apache Spark 3.1.0, 
```

Comment:
This says it will run tests against Spark 3.1.0 but only runs one test.  Either fix the description or fix the example.